### PR TITLE
Add: Container Signing action

### PR DESCRIPTION
**Pull request by pascalholthaus**
## What
Add: Container Signing action

## Why
DEVOPS-574

## References
DEVOPS-574

### CODE REVIEW

File: container-signing/action.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - uses: sigstore/cosign-installer@main
+    - name: Sign container image (public/private keypair)
+      if: ${{ inputs.cosign-key-password && inputs.cosign-key }}
+      shell: bash
+      env:
+        COSIGN_PASSWORD: ${{ inputs.cosign-key-password }}
+        COSIGN_KEY: ${{ inputs.cosign-key }}
+      run: |
+        echo "${{ inputs.docker-tags }}" | \
+          xargs -I {} cosign sign --recursive --key env://COSIGN_KEY {}@${{ inputs.docker-digest }}
```
Comment (bjoernricks) [inline container-signing/action.yaml:34]:
```suggestion
        echo "${{ inputs.image-tags }}" | \
          xargs -I {} cosign sign --recursive --key env://COSIGN_KEY {}@${{ inputs.image-digest }}
```

File: container-signing/action.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+inputs:
+  docker-digest:
+    description: "Set the digest from the docker build and push action e.g the output of steps.build-and-push.outputs.digest."
+    required: true
+  docker-tags:
+    description: "Set the tags from the docker meta action e.g the output of steps.meta.outputs.tags."
+    required: true
+  cosign-key-password:
+    description: "Set the cosign key password, it not set a keyless signature will be created."
```
Comment (bjoernricks) [inline container-signing/action.yaml:12]:
```suggestion
    description: "Set the cosign key password, if not set a keyless signature will be created."
```

File: container-signing/action.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    required: false
+  cosign-key:
+    description: "Set the cosign key, it not set a keyless signature will be created."
```
Comment (bjoernricks) [inline container-signing/action.yaml:15]:
```suggestion
    description: "Set the cosign key, if not set a keyless signature will be created."
```

File: container-signing/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Container signing
+
+Cosign based action to create container signatures.
```
Comment (bjoernricks) [inline container-signing/README.md:3]:
```suggestion
Cosign based action to create container signatures. Depending on the input public/private key or keyless signatures are created.
```

File: container-signing/action.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    required: false
+
+branding:
+  icon: "package"
+  color: "green"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: sigstore/cosign-installer@main
```
Comment (bjoernricks) [inline container-signing/action.yaml:25]: IMHo we should use the latest release version.
```suggestion
    - uses: sigstore/cosign-installer@v3
```

File: container-signing/action.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        echo "${{ inputs.docker-tags }}" | \
+          xargs -I {} cosign sign --recursive --key env://COSIGN_KEY {}@${{ inputs.docker-digest }}
+    - name: Sign the published Container Image (Keyless)
+      if: ${{ !inputs.cosign-key-password || !inputs.cosign-key }}
+      shell: bash
+      env:
+        COSIGN_EXPERIMENTAL: "true"
+      run: |
+       echo "${{ inputs.docker-tags }}" | \
+         xargs -I {} cosign sign --recursive {}@${{ inputs.docker-digest }}
```
Comment (bjoernricks) [inline container-signing/action.yaml:42]:
```suggestion
       echo "${{ inputs.image-tags }}" | \
         xargs -I {} cosign sign --recursive {}@${{ inputs.image-digest }}
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: container-signing/README.md (hunk 4 above)
Comment (pascalholthaus) [reply container-signing/README.md:3]: done

File: container-signing/README.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+## Action Configuration
+
+|Input Variable|Description|Optional|
+|--------------|-----------|--------|
+|image-tags|Set the tags from the docker meta action e.g the output of steps.meta.outputs.tags.|false|
+|image-digest|Set the digest from the docker build and push action e.g the output of steps.build-and-push.outputs.digest.|true|
+|cosign-key-password|Set the cosign key password, if not set a keyless signature will be created.|true|
+|cosign-key|Set the cosign key, if not set a keyless signature will be created.|true|
```
Comment (bjoernricks) [inline container-signing/README.md:41]:
```suggestion
|Input Variable|Description| |
|--------------|-----------|--------|
|image-tags|Set the tags from the docker meta action e.g the output of steps.meta.outputs.tags.|Required|
|image-digest|Set the digest from the docker build and push action e.g the output of steps.build-and-push.outputs.digest.|Optional|
|cosign-key-password|Set the cosign key password, if not set a keyless signature will be created.|Optional|
|cosign-key|Set the cosign key, if not set a keyless signature will be created.|Optional|
```

**Review by bjoernricks — APPROVED** [review]

**Review by bjoernricks — APPROVED** [review]

**Review by bjoernricks — DISMISSED** [review]

**Review by y0urself — APPROVED** [review]

**Pull request merged**